library/core/src/ptr/const_ptr.rs

```diff
@@ -792,11 +795,14 @@ impl<T: ?Sized> *const T {
     #[unstable(feature = "const_raw_ptr_comparison", issue = "53020")]
     #[rustc_const_unstable(feature = "const_raw_ptr_comparison", issue = "53020")]
     #[inline]
-    pub const fn guaranteed_eq(self, other: *const T) -> bool
+    pub const fn guaranteed_eq(self, other: *const T) -> Option<bool>
```

Comment:
The doc comment needs updating, it talks about spuriously returning `false`.
(Also applies to all the other public methods.)